assets/js/docs.js

```diff
@@ -78,30 +79,40 @@
           content = this.state.content,
           searchValue = this.state.searchValue;
 
-      var filtered = _(content)
-        .map(function(collection) {
-          // If search is for a collection title, return collection.
-          if (search(collection.title, searchValue)) {
-            return collection;
-          }
-          // If search is for a function, return matching functions.
-          return {
-            'title': collection.title,
-            'expanded': collection.expanded,
-            'functions': _.filter(collection.functions, function(data) {
-              return search(data.name, searchValue);
-            })
-          };
-        })
-        .reject(function(collection) {
-          return _.isEmpty(collection.functions);
-        })
-        .value();
+      var filtered =
+        _(content)
+          .map(function(collection) {
+            var result = {
+              'title': collection.title,
+              'expanded': collection.expanded,
+              'visible': search(collection.title, searchValue),
+              'functions': _.map(collection.functions, function(data) {
+                data.visible = search(collection.title, searchValue) || search(data.name, searchValue);
+                return data;
+              })
+            };
+
+            // The collection is visible if searchValue matches the title
+            // or if there are visible functions
+            result.visible =
```

Comment:
@jdalton Would love some feedback on how to make this cleaner. Don't like assigning twice.